extensions/azure-keyvault/runtime/src/main/java/io/quarkiverse/azure/keyvault/secret/runtime/KeyVaultSecretClientProducer.java

```diff
@@ -18,29 +19,29 @@ public class KeyVaultSecretClientProducer {
 
     @Produces
     public SecretClient createSecretClient() {
-        if (!secretConfiguration.enabled) {
+        if (!secretConfiguration.enabled()) {
             return null;
         }
 
-        assert secretConfiguration.endpoint.isPresent() : "The endpoint of Azure Key Vault Secret must be set";
+        assert secretConfiguration.endpoint().isPresent() : "The endpoint of Azure Key Vault Secret must be set";
```

Comment:
Now we support multiple key vaults, maybe `endpoint` can be optional and return null if it's not provided?

Reply:
The `endpoint` is optional as the `enabled` property requires.

Reply:
What I meant is that even though `enabled` is true, `endpoint` can still be empty as now we support multiple key vaults.

Reply:
Yes, the property can be left unset. The user can use `kv//kv-name/secret` to access Key Vault Secret value.

Reply:
Let me clarify my question: `assert secretConfiguration.endpoint().isPresent() : "The endpoint of Azure Key Vault Secret must be set";` requires user to provide `endpoint` if `enabled` is `true`. Is it what we want? In other words, user always has to provide a so-called default key vault endpoint, even though its secrets are not used as config properties in the app. Right?

Reply:
endpoint is required for cases:
- Inject SecretClient and AsyncClient
- Use path like ${kv//<secre-name>} ir ${kv//<secret-name>/versions/<version-id>}

endpoint is not required for other path that specifying kv-name, e.g. ${kv//<kv-name>/...}

Use endpoint to specify a default key vault is recommended. So the document set the property in the sample application.